src/Microsoft.Azure.WebJobs.Host/Scale/ScaleLoggerExtension.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    public static class ScaleLoggerExtension
+    {
+        private static readonly EventId FunctionScaleErrorEventId = new EventId(8001, "FunctionScaleError");
+        private static readonly EventId LogFunctionScaleVoteEventId = new EventId(8002, "LogFunctionScaleVote");
+        private static readonly string FunctionNameKey = "functionName";
+        private static readonly string TargetWorkerCountKey = "targetWorkerCount";
+        private static readonly string QueueLengthKey = "queueLength";
+        private static readonly string ConcurrencyKey = "concurrency";
+        private static readonly string VoteKey = "vote";
+        private static readonly string ReasonKey = "reason"; // Renamed from 'Reason'
+
+        // <summary>
+        // Logs a scale vote for a function with the specified name, vote, and optional reason.
+        // </summary>
+        public static void LogFunctionScaleVote(this ILogger logger, string functionName, string vote, string reason = null)
+        {
+            using (logger.BeginScope(new Dictionary<string, object>
```

Comment:
If you include all the parameters in the template, the scope will not be necessary. In your message below, `functionName` and `vote` will already become log properties. All you would need is to add `reason` to it.

And then if you _really_ don't want to add `reason` to the message, you can define your own struct which implements `IEnumerable<KeyValuePair<string, object>>`. The contents of enumeration are your properties, and then the formatter you pass in to the `Log` call is your message.

Reply:
We want to add all values to the scope as those values need to be extracted in SC and logged as separate columns.